### PR TITLE
ci: use actions-setup-redis

### LINKED ISSUE #118: ci: avoid building Redis everytime
- #113
- https://github.com/actions/cache
- https://github.com/shogo82148/actions-setup-redis
  - Used by [node-redis](https://github.com/NodeRedis/node-redis)

### PR DESCRIPTION
**Pull request by tumile**
Closes #118.

- [x] Use [actions-setup-redis](https://github.com/shogo82148/actions-setup-redis) in CI.
- [x] With redis setup moved to action the Makefile does not have a much meaningful purpose except for lint and test commands. So I moved the commands into `build.yml` as well.
- [x] Set deno version in `build.yml` along with redis version.
- [x] Moved `redis.conf` into `test/server` to cleanup root-level dir.

Result: [CI finished in 34s](https://github.com/denolib/deno-redis/actions/runs/174826786) instead of 2m

### CODE REVIEW

**Comment by tumile** [comment]
pinging @uki00a for review

File: .denov [hunk 1]
```diff
@@ -1 +0,0 @@
-v1.2.0
```
Comment (uki00a) [inline .denov:1]: `.denov` is used by [dink-bot](https://github.com/keroxp/dink-bot) through `.github/workflows/bump.yml`.
Could you revert this for now?

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -1,33 +1,45 @@
...
   pull_request:
     branches:
       - "**"
+
 jobs:
   build:
     runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        deno-version: [1.x]
```
Comment (uki00a) [inline .github/workflows/build.yml:16]: Is it possible to change this to use `.denov`?

File: .denov (hunk 1 above)
Comment (tumile) [reply .denov:1]: I see, I'll do that

File: .github/workflows/build.yml (hunk 2 above)
Comment (tumile) [reply .github/workflows/build.yml:16]: Hmm I don't think so because `cat .denov` needs to be run by `run` command

**Comment by tumile** [comment]
Updated!

**Comment by uki00a** [comment]
Merged, thanks @tumile!

**Pull request merged**